--- linux-3.10.0-327.el7.centos.x86_64/Makefile.orig	2015-10-29 15:56:51.000000000 -0500
+++ linux-3.10.0-327.el7.centos.x86_64/Makefile	2015-11-21 09:49:08.038404090 -0600
@@ -383,7 +383,7 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstric
 
 ifneq ($(WITH_GCOV),1)
 ifeq ($(KBUILD_EXTMOD),)
-ifneq (,$(filter $(ARCH), x86 x86_64 powerpc))
+ifneq (,$(filter $(ARCH), x86_64 powerpc))
 KBUILD_CFLAGS   += $(call cc-ifversion, -eq, 0408, -Werror)
 endif
 # powerpc is compiled with -O3 and gcc 4.8 has some known problems
